client/v2/tx/builder.go

```diff
@@ -0,0 +1,35 @@
+package tx
+
+import (
+	txv1beta1 "cosmossdk.io/api/cosmos/tx/v1beta1"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+type ExtendedTxBuilder interface {
+	SetExtensionOptions(extOpts ...*codectypes.Any)
+}
+
+// TxBuilder defines an interface which an application-defined concrete transaction
+// type must implement. Namely, it must be able to set messages, generate
+// signatures, and provide canonical bytes to sign over. The transaction must
+// also know how to encode itself.
+type TxBuilder interface {
+	GetTx() txv1beta1.Tx
+	Sign() error
+	SetMsgs(msgs ...*anypb.Any) error
+	SetMemo(memo string)
+	SetFeeAmount(amount txv1beta1.Fee)
+	SetFeePayer(feePayer string)
+	SetGasLimit(limit uint64)
+	SetTimeoutHeight(height uint64)
+	SetFeeGranter(feeGranter string)
+	SetUnordered(v bool)
+	SetSignatures(doc txv1beta1.SignDoc) error
+	SetAuxSignerData(data txv1beta1.AuxSignerData) error
+}
```

Comment:
Input types have been replaced with the corresponding ones in `cosmos/api`.
**Problem**: many functions expects the `sdk.Types` version of these types instead of the api ones, for example `SetSignatures` originally has as input [SignatureV2](https://github.com/cosmos/cosmos-sdk/blob/8f0d5b15f0b10da7645d7fc1aa868fe44e3f3a44/types/tx/signing/signature.go#L15) but now it uses the `SignDoc` type (might be the incorrect type for this case but still the problem is the same) IMO this will force to every implementation of `TxBuilder` to have especial code to convert wrapper structs (`SignatureV2`, `Tx`, `Msg`, etc) to its corresponding protobuf equivalent and deal with the compatibility for protov1 and protov2 types.

`Idea`: Since one of the requirements of this epic is to fully support  protov2 types, adapters that convert the `sdk.Types` to the protoV2 type could be a way to facilitate this. But also to keep compatibility with protov1 these adapters should also know how to convert `sdk.Type` to protov1 (?)